2008-03-03  Ulrich Drepper  <drepper@redhat.com>

	[BZ #5854]
	* nis/ypclnt.c (yp_order): Fix handling of return value of
	do_ypcall_tr call.
	Patch by Jeff Moyer <jmoyer@redhat.com>.

--- libc/nis/ypclnt.c	28 Apr 2006 16:59:22 -0000	1.58
+++ libc/nis/ypclnt.c	4 Mar 2008 00:27:41 -0000	1.59
@@ -634,7 +634,7 @@ yp_order (const char *indomain, const ch
 			 (caddr_t) &req, (xdrproc_t) xdr_ypresp_order,
 			 (caddr_t) &resp);
 
-  if (result == YPERR_SUCCESS)
+  if (result != YPERR_SUCCESS)
     return result;
 
   *outorder = resp.ordernum;
